modules/apps/batch-planner/batch-planner-web/src/main/resources/META-INF/resources/js/import/ImportPreviewModal.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayModal from '@clayui/modal';
+import ClayTable from '@clayui/table';
+import React from 'react';
+
+import CellPreview from './CellPreview';
+
+const ImportPreviewModal = ({
+	closeModal,
+	fieldsSelections,
+	fileContent,
+	handleEditCell,
+	setStartImport,
+}) => {
+	const fileFieldsToMap = fieldsSelections
+		? Object.values(fieldsSelections)
+		: {};
+
+	return (
+		<>
+			<ClayModal.Header>
+				{Liferay.Language.get('preview')}
+			</ClayModal.Header>
+
+			<ClayModal.Body>
+				<ClayTable>
+					<ClayTable.Head>
+						<ClayTable.Row>
+							{Array.from(fileFieldsToMap).map(
+								(element, index) => {
+									let label;
+									if (isNaN(parseInt(element, 10))) {
```

Comment:
When, after the changes, the component receives the data formatted the right way, this logic won't be needed